File: capture_screenshot.js

import puppeteer from 'puppeteer';

(async () => {
    try {
        const browser = await puppeteer.launch({
            headless: "new",
            args: ['--no-sandbox', '--disable-setuid-sandbox']
        });
        const page = await browser.newPage();

        // Capture Console Logs
        page.on('console', msg => console.log('PAGE LOG:', msg.text()));
        page.on('pageerror', error => console.error('PAGE ERROR:', error.message));
        page.on('requestfailed', request => console.error('FAILED REQUEST:', request.url(), request.failure().errorText));

        console.log('Navigating to http://localhost:4200...');
        await page.setViewport({ width: 1280, height: 800 });

        // Increased timeout and changed waitUntil to networkidle2 to be more robust
        await page.goto('http://localhost:4173', { waitUntil: 'networkidle2', timeout: 30000 });

        console.log('Taking screenshot...');
        await page.screenshot({ path: 'screenshot_error_debug.png', fullPage: true });

        await browser.close();
        console.log('Screenshot saved to screenshot_error_debug.png');
    } catch (error) {
        console.error('Error executing script:', error);
        process.exit(1);
    }
})();

File: capture_screenshot.py
import os
from playwright.sync_api import sync_playwright

# Set HOME environment variable explicitly to avoid the error
os.environ["HOME"] = os.path.expanduser("~")

def take_screenshot():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        print("Navigating to http://localhost:4173...")
        page.goto("http://localhost:4173")
        page.wait_for_load_state("networkidle")
        print("Taking screenshot...")
        page.screenshot(path="screenshot_preview.png", full_page=True)
        browser.close()
        print("Screenshot saved to screenshot_preview.png")

if __name__ == "__main__":
    take_screenshot()
